components/search/src/types.rs

```diff
@@ -165,8 +174,8 @@ pub struct SearchEngineDefinition {
     pub name: String,
 
     /// The partner code for the engine. This will be inserted into parameters
-    /// which include `{partnerCode}`.
-    pub partner_code: Option<String>,
+    /// which include `{partnerCode}`. May be the empty string.
+    pub partner_code: String,
```

Comment:
I made this change, as I think we'd generally replace `{partnerCode}` with the string, whether it is empty or not.